### PR TITLE
PT-161582967 - Intermittent test failure in TX GC

### PR DESCRIPTION
**Pull request by hanssv**
Remove unused code, make stop_generation synchronous and harden the test

### CODE REVIEW

**Comment by lucafavatella** [comment]
Tests [failed on CI](https://circleci.com/gh/aeternity/epoch/41063) for unrelated failure - I am taking the liberty to restart job.
```
%%% aehttp_integration_SUITE ==> all.external_endpoints.pending_transactions: FAILED
%%% aehttp_integration_SUITE ==> {{badmatch,{409999999999999999872,409999999999999979872}},
 [{aehttp_integration_SUITE,pending_transactions,1,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,3001}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1062}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,994}]}]}
```

File: apps/aecore/test/aecore_txs_SUITE.erl [hunk 1]
```diff
@@ -129,34 +120,50 @@ txs_gc(Config) ->
...
+    %% Now if at height 5 or 6 there should be 2 transactions in mempool
     case Height3 of
-        6                -> {2, _} = {length(Txs4), Txs4};
+        HH3 when HH3 =<6 -> pool_check(N1, 2);
         HH3 when HH3 > 6 -> ct:log("Skipping assert; micro fork advanced height to far...")
     end,
 
-    %% Mine 1 more key block if Height is 6 ensure one TX should be GC:ed.
-    [ aecore_suite_utils:mine_key_blocks(N1, 1) || Height3 == 6 ],
+    %% Mine 1 more key block if Height is 5 ensure one TX should be GC:ed.
```
Comment (lucafavatella) [inline apps/aecore/test/aecore_txs_SUITE.erl]: > 1 more key block if Height is 5

may need update though not required.

File: apps/aecore/src/aec_tx_pool.erl [hunk 2]
```diff
@@ -372,11 +372,6 @@ int_get_candidate_fold(Gas, GasLimit, Db, Dbs = #dbs{}, {Txs, Cont},
                        AccountsTree, Height, Acc) when Gas > GasLimit ->
     {RemGas, NewAcc} = fold_txs(Txs, Gas, GasLimit, Db, Dbs,
                                 AccountsTree, Height, Acc),
-        lists:foldl(
-          fun(T, {Gas1, Acc1}) ->
-                  int_get_candidate_(T, Gas1, Db, Dbs, AccountsTree,
```
Comment (lucafavatella) [inline apps/aecore/src/aec_tx_pool.erl]: This has side effect in move_to_visited though seems genuine duplicated code.

Comment (hanssv) [reply apps/aecore/src/aec_tx_pool.erl]: Yes, that is how I noticed (we were adding it to GC twice!)...

File: apps/aecore/src/aec_conductor.erl [hunk 3]
```diff
@@ -227,6 +227,7 @@ handle_call({post_block, Block},_From, State) ->
     {reply, Reply, State1};
 handle_call(stop_mining,_From, State = #state{ consensus = Cons }) ->
     epoch_mining:info("Mining stopped"),
+    aec_block_generator:stop_generation(),
```
Comment (lucafavatella) [inline apps/aecore/src/aec_conductor.erl]: Please ensure that merging to master this does not introduce [double stopping](https://github.com/aeternity/epoch/blob/28be867f7f3e7bfb66eaa3c0a487933815d83f5a/apps/aecore/src/aec_conductor.erl#L230-L231).

**Comment by hanssv** [comment]
Pushed a rebased version to avoid stopping twice cc @lucafavatella  :-)

**Pull request merged**